proxy/invoke.go

```diff
@@ -26,13 +27,13 @@ func InvokeFunction(gateway string, name string, bytesIn *[]byte, contentType st
 	qs, qsErr := buildQueryString(query)
 	if qsErr != nil {
 		return nil, qsErr
-
 	}
 
 	gatewayURL := gateway + "/function/" + name + qs
 	// fmt.Println(gatewayURL)
 	req, _ := http.NewRequest(http.MethodPost, gatewayURL, reader)
 	req.Header.Add("Content-Type", contentType)
+	SetAuth(req, gateway)
```

Comment:
This is good for now.

But I think we will want to be more nuanced with locking down endpoints, with functions and system being secured using different credentials (and most likely with individual functions with their own auth (or none)). (^ @alexellis)

Reply:
Sending basic authentication when the function is open will result in the header being ignored, so we should be ok.